src/test/scala/org/broadinstitute/dsde/firecloud/webservice/TrialRegistrationSpec.scala

```diff
@@ -0,0 +1,140 @@
+package org.broadinstitute.dsde.firecloud.webservice
+
+import org.broadinstitute.dsde.firecloud.{FireCloudException, FireCloudExceptionWithErrorReport}
+import org.broadinstitute.dsde.firecloud.dataaccess.{MockSamDAO, MockThurloeDAO}
+import org.broadinstitute.dsde.firecloud.dataaccess.MockThurloeDAO._
+import org.broadinstitute.dsde.firecloud.mock.MockUtils.randomAlpha
+import org.broadinstitute.dsde.firecloud.model.ErrorReportExtensions.FCErrorReport
+import org.broadinstitute.dsde.firecloud.model.{BasicProfile, RegistrationInfo, Trial, UserInfo, WithAccessToken, WorkbenchEnabled, WorkbenchUserInfo}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.{impBasicProfile, impRegistrationInfo}
+import org.broadinstitute.dsde.firecloud.model.Trial.TrialStates._
+import org.broadinstitute.dsde.firecloud.model.Trial.UserTrialStatus
+import org.broadinstitute.dsde.firecloud.service.{BaseServiceSpec, FireCloudRequestBuilding, RegisterService}
+import org.broadinstitute.dsde.rawls.model.RawlsUserEmail
+import spray.http.HttpResponse
+import spray.http.StatusCodes.{NotFound, OK}
+import spray.httpx.SprayJsonSupport
+import spray.json.DefaultJsonProtocol
+
+import scala.concurrent.Future
+import scala.util.Try
+
+class TrialRegistrationSpec extends BaseServiceSpec with RegisterApiService with FireCloudRequestBuilding
+  with SprayJsonSupport with DefaultJsonProtocol {
+
+  def actorRefFactory = system
+
+  val regSamDao = new UserRegistrationSamDAO()
+  val regThurloeDao = new UserRegistrationThurloeDAO()
+
+  val testApp = app.copy(samDAO = regSamDao, thurloeDAO = regThurloeDao)
+
+  val registerServiceConstructor: () => RegisterService = RegisterService.constructor(testApp)
+
+  val registerPath = "/register/profile"
+
+  val fullProfile = BasicProfile(
+    firstName= randomAlpha(),
+    lastName = randomAlpha(),
+    title = randomAlpha(),
+    contactEmail = Option.empty,
+    institute = randomAlpha(),
+    institutionalProgram = randomAlpha(),
+    programLocationCity = randomAlpha(),
+    programLocationState = randomAlpha(),
+    programLocationCountry = randomAlpha(),
+    pi = randomAlpha(),
+    nonProfitStatus = randomAlpha()
+  )
+
+  "When updating the profile for a pre-existing, registered user" - {
+    "should not read or write trial status" in {
+      Post(registerPath, fullProfile) ~> dummyUserIdHeaders(NORMAL_USER, email = NORMAL_USER) ~> sealRoute(registerRoutes) ~> check {
+        status should equal(OK)
+        val regInfo = responseAs[RegistrationInfo]
+        // if NORMAL_USER triggers a call to either getTrialStatus or saveTrialStatus, UserRegistrationThurloeDAO will
+        // fail and RegistrationInfo will have a message below.
+        assert(regInfo.message.isEmpty, "- if this reports a BoxedError it is likely the assert inside UserRegistrationThurloeDAO.getTrialStatus or saveTrialStatus")
```

Comment:
I find the "if this reports a BoxedError" distasteful - but it's the result of an `assert` failing and throwing an exception inside a Future - the Future is in the mock DAOs at the bottom of this file. I didn't want to rewrite the runtime Future-handling and recover-handling - so I declared this good enough for the tests.